src/identity-utils.ts

```diff
@@ -24,16 +24,20 @@ export const cacheOrClearIdCache = (
     method: string,
     knownIdentities: IKnownIdentities,
     idCache: BaseVault<Dictionary<ICachedIdentityCall>>,
-    xhr: XMLHttpRequest
+    xhr: XMLHttpRequest,
+    parsingCachedResponse: boolean,
 ): void => {
+    // when parsing a response that has already been cached, simply return instead of attempting another cache
+    if (parsingCachedResponse) { return; }
+
     const CACHE_HEADER = 'x-mp-max-age';
 
     // default the expire timestamp to one day in milliseconds unless a header comes back
-    let expireTimestamp = new Date().getTime() + ONE_DAY_IN_SECONDS * MILLIS_IN_ONE_SEC;
-
+    let now = new Date().getTime();
+    let expireTimestamp = now + ONE_DAY_IN_SECONDS * MILLIS_IN_ONE_SEC;
```

Comment:
Originally I was just setting the timestamp to what was in the header, which is just 1 day in milliseconds, so it was firing the identity calls all the time since the logic was such that it would fire if the timestamp expires, which it always does.